nixos/modules/services/networking/seafile.nix

```diff
@@ -0,0 +1,229 @@
+{ config, lib, pkgs, ...}:
+
+with lib;
+
+let
+  cfg = config.services.seafile;
+
+  seafileIni = pkgs.writeText "seafile.ini" ''
+    ${seafileDataDir}
+  '';
+
+  seahubSettings = pkgs.writeText "seahub_settings.py" ''
+    FILE_SERVER_ROOT = '${cfg.serviceUrl}/seafhttp'
+    DATABASES = {
+        'default': {
+            'ENGINE': 'django.db.backends.sqlite3',
+            'NAME': '${seahubDir}/seahub.db',
+        }
+    }
+    MEDIA_ROOT = '${seahubDir}/media/'
+    THUMBNAIL_ROOT = '${seahubDir}/thumbnail/'
+    ${cfg.seahubExtraConf}
+  '';
+
+  centralConfigDir = "${cfg.dataDir}/conf";
+  ccnetDir = "${cfg.dataDir}/ccnet";
+  seafileDataDir = "${cfg.dataDir}/seafile-data";
+  seahubDir = "${cfg.dataDir}/seahub";
+  runDir = "/run/seafile";
+  logDir = "${cfg.dataDir}/log";
+
+in {
+
+  ###### Interface
+
+  options.services.seafile = {
+    enable = mkEnableOption ''
```

Comment:
I believe this will render poorly. The `mkEnableOption` helper expects simply the name of the service; if you wish to provide a longer form description, you must override it (as in `mkEnableOption "foo" // { description = "My long description"; }`).